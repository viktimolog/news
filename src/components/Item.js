import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import {Link} from 'react-router'
import './style.css'

const Styles = {
    display: 'flex',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
    width: '30%',
    margin: 10,
    padding: 10,
    paddingBottom: 6,
    border: '1px',
    borderStyle: 'solid',
    borderColor: '#ff829e'
}

const Item = ({item}) => (
    <Card
        // className="item"
        style={Styles}
    >
        <CardMedia className="media">
            <img src={item.general.avatar} alt=""/>
        </CardMedia>
        <CardContent>
            <Typography gutterBottom variant="headline" component="h2">
                {item.general.firstName} {item.general.lastName}
            </Typography>
            <Typography component="p">
                {item.job.title}
            </Typography>
        </CardContent>
        <CardActions>
            <Button
                style={{margin: -14}}
                size="small"
                color="primary"
                variant="contained"
            >
                <Link to={`/items/${item.id}`} style={{color: 'white'}}>More</Link>
            </Button>
        </CardActions>
    </Card>
)

Item.propTypes = {
    item: PropTypes.object.isRequired,
}

export default Item