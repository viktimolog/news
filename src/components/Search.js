import React from 'react'
import { Input } from 'semantic-ui-react'

export default class Search extends React.Component {
  state = {text: ''}

  handleTextChange = event => {
    this.setState({
      text: event.target.value.trim()
    })
    this.props.findItems(event.target.value.trim())
  }

  render () {
    return (
      <Input
        style={{width: '100%'}}
        icon='search'
        value={this.state.text}
        onChange={this.handleTextChange}
      />)
  }
}