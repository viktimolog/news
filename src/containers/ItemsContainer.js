import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Items from '../components/Items'
// import Search from '../components/Search'
import { getItems, setCurItem, findItems } from '../actions/actionCreator'

class ItemsContainer extends React.Component {

  componentDidMount () {
    this.props.getItems()
  }

  render () {
    if (this.props.curItem !== false) return null
    return (
      <div>
        {/*<Search findItems={this.props.findItems}/>*/}
        <Items
          items={this.props.items}
          setCurItem={this.props.setCurItem}/>
      </div>)
  }
}

ItemsContainer.propTypes = {
  curItem: PropTypes.oneOfType([PropTypes.Boolean, PropTypes.object]).isRequired,
  items: PropTypes.array.isRequired,
  setCurItem: PropTypes.func.isRequired,
  getItems: PropTypes.func.isRequired,
  findItems: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  items: state.mainReducer.items,
  curItem: state.mainReducer.curItem
})

const mapDispatchToProps = {
  setCurItem,
  getItems,
  findItems
}

export default connect(mapStateToProps, mapDispatchToProps)(ItemsContainer)

