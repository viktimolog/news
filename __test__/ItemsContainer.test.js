import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import ItemsContainer from '../src/containers/ItemsContainer'
import Item from '../src/components/Item'
import Search from '../src/components/Search'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// Snapshot for Item React Component
describe('>>>Item --- Snapshot', () => {
  it('+++capturing Snapshot of Home', () => {
    const renderedValue = renderer.create(<Item item={
      {
        'id': '0',
        'general': {
          'firstName': 'Liana',
          'lastName': 'Crooks',
          'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg'
        },
        'job': {
          'company': 'Ledner, Johnson and Predovic',
          'title': 'Investor Functionality Coordinator'
        },
        'contact': {
          'email': 'Gerry_Hackett77@gmail.com',
          'phone': '(895) 984-0132'
        },
        'address': {
          'street': '1520 Zemlak Cove',
          'city': 'New Devon',
          'zipCode': '42586-7898',
          'country': 'Guinea-Bissau'
        }
      }
    }

    />).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })
})

describe('>>>ItemsContainer --- REACT-REDUX (Shallow + passing the {store} directly)', () => {
  const mockStore = configureStore();
  let store, container;

  const initialState = {
    items: [
      {
        "id": "0",
        "general": {
          "firstName": "Liana",
          "lastName": "Crooks",
          "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg"
        },
        "job": {
          "company": "Ledner, Johnson and Predovic",
          "title": "Investor Functionality Coordinator"
        },
        "contact": {
          "email": "Gerry_Hackett77@gmail.com",
          "phone": "(895) 984-0132"
        },
        "address": {
          "street": "1520 Zemlak Cove",
          "city": "New Devon",
          "zipCode": "42586-7898",
          "country": "Guinea-Bissau"
        }
      }
      ]
  };


  beforeEach(() => {
    store = mockStore(initialState);
    container =
      shallow(<ItemsContainer store={store} />);
  });

  it('+++ render the connected(SMART) component', () => {
    expect(container.length).toEqual(1);
  });

  it('+++ check Prop matches with initialState', () => {
    expect(container.prop('items')).toEqual(initialState.items);
  });
});

