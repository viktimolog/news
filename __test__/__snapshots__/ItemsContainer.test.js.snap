// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Item --- Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  style={
    Object {
      "alignItems": "center",
      "border": "1px",
      "borderColor": "#ff829e",
      "borderStyle": "solid",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
      "paddingBottom": 6,
      "width": "30%",
    }
  }
>
  <div
    className="MuiCardMedia-root-29 media"
  >
    <img
      alt=""
      src="https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg"
    />
  </div>
  <div
    className="MuiCardContent-root-31"
  >
    <h2
      className="MuiTypography-root-32 MuiTypography-headline-37 MuiTypography-gutterBottom-49"
    >
      Liana
       
      Crooks
    </h2>
    <p
      className="MuiTypography-root-32 MuiTypography-body1-41"
    >
      Investor Functionality Coordinator
    </p>
  </div>
  <div
    className="MuiCardActions-root-56"
  >
    <button
      className="MuiButtonBase-root-82 MuiButton-root-58 MuiButton-contained-67 MuiButton-containedPrimary-68 MuiButton-raised-70 MuiButton-raisedPrimary-71 MuiButton-sizeSmall-79 MuiCardActions-action-57"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "margin": -14,
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-59"
      >
        <a
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          More
        </a>
      </span>
      <span
        className="MuiTouchRipple-root-85"
      />
    </button>
  </div>
</div>
`;
